src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx

```diff
@@ -366,6 +379,54 @@ function HistogramViz(props: VisualizationProps) {
         }
       : undefined;
 
+  // custom legend items for checkbox
+  const legendItems: LegendItemsProps[] = useMemo(() => {
+    const legendData = !isFaceted(data.value)
+      ? data.value?.series
+      : data.value?.facets[0].data.series;
+
+    return legendData != null
+      ? legendData
+          .map((dataItem: HistogramDataSeries, index: number) => {
+            return {
+              label: dataItem.name,
+              // histogram plot does not have mode, so set to square for now
+              marker: 'square',
+              markerColor:
+                dataItem.name === 'No data'
+                  ? '#E8E8E8'
+                  : ColorPaletteDefault[index],
+              // deep comparison is required for faceted plot
+              hasData: !isFaceted(data.value) // no faceted plot
+                ? dataItem.bins != null && dataItem.bins.length > 0
+                  ? true
+                  : false
+                : data.value?.facets // faceted plot: here data.value is full data
+                    .map(
+                      (el: { label: string; data: HistogramData }) =>
+                        el.data.series[index].bins != null &&
+                        el.data.series[index].bins.length > 0
+                    )
+                    .includes(true)
+                ? true
+                : false,
+              group: 1,
+              rank: 1,
+            };
+            // histogram viz uses stack option so reverse the legend order!
+          })
+          .reverse()
+      : [];
+  }, [data]);
+
+  // use this to set all checked
+  useEffect(() => {
```

Comment:
I thought I had a simple solution to the problem of this overwriting the saved configuration, but it wasn't so simple.

So are we going to solve this separately?  Don't forget to make a ticket if so.

Reply:
will make a ticket to handle saving/restoring checked legend items. Actually legend markers are for both boxplot and scatter plot as boxplot utilized different marker style unlike histogram etc.: square but light color and border for boxplot ones.

Reply:
As pointed out by Dave, we already have saving/restoring ticket made long time ago, #172, so I don't need to make it: . just assigned it to me so that I won't forget 😃 Considering other tasks I have to do for b55, I will defer it, though.